### PR TITLE
Add zonefile download

### LINKED ISSUE #2618: DNS DKIM entry is broken up in two parts (for 2048b keys), making it error-prone
### Environment

- [x] docker-compose
- [ ] kubernetes
- [ ] docker swarm

### Version

- Version: 1.9

## Description

DNS DKIM entry in mailu-admin is broken into two parts for 2048b keys.

## Replication Steps

1. Set up a mail domain
2. Generate keys in admin
3. Observe DNS DKIM entry

## Observed behaviour

```
[...] 600 IN TXT "v=DKIM1; k=rsa; p=[.....]" "[......]"
```

(the " " in between is hard to notice, and easily make people just copy that into the dns configuration, rendering the dkim entry invalid)

## Expected behaviour

```
[...] 600 IN TXT "v=DKIM1; k=rsa; p=[.....][......]"
```

(shouldn't be broken up into two parts)

## Logs

not relevant

### PR DESCRIPTION
**Pull request by MajliTech**
## What type of PR?

Feature

## What does this PR do?
On /admin/domain/details/{domain}, adds a button to redirect to /admin/domain/details/{domain}/zonefile, which downloads a zonefile for this domain.

closes #2618

## Prerequisites
Before we can consider review and merge, please make sure the following list is done and checked.
If an entry in not applicable, you can check it or remove it from the list.

- [X] In case of feature or enhancement: documentation updated accordingly
- [X] Unless it's docs or a minor change: add [changelog](https://mailu.io/master/contributors/workflow.html#changelog) entry file.

### CODE REVIEW

**Comment by nextgens** [comment]
@ghostwheel42 maybe that's the way forward to handle #2618 ? With it we could also remove the record splitting

File: core/admin/mailu/ui/views/domains.py [hunk 1]
```diff
@@ -70,6 +70,22 @@ def domain_details(domain_name):
...
+    final = final + domain.dns_spf+"\n"
+    if domain.dkim_publickey:
+        final = final + domain.dkim_publickey+"\n"
+        final = final + domain.dns_dkim+"\n"
+        final = final + domain.dns_dmarc+"\n"
+    if domain.dns_tlsa:
+        final = final + domain.dns_tlsa
+    for i in domain.dns_autoconfig:
+        final = final + i+"\n"
+    return flask.Response(final,content_type="text/plain",headers={'Content-disposition': 'attachment; filename='+domain.name+'-zonefile.txt'})
```
Comment (ghostwheel42) [inline core/admin/mailu/ui/views/domains.py:87]: LGTM.

Can you create the zone in a more pythonic way like this?

```suggestion
    res = [domain.dns_mx, domain.dns_spf]
    if domain.dkim_publickey:    
        res.extend([domain.dkim_publickey, domain.dns_dkim, domain.dns_dmarc])
    if domain.dns_tlsa:
        res.append(domain.dns_tlsa)
    res.extend(domain.dns_autoconfig)
    res.append("")
    return flask.Response("\n".join(res), content_type="text/plain", headers={"Content-disposition": f"attachment; filename={domain.name}-zonefile.txt"})
```

Comment (ghostwheel42) [reply core/admin/mailu/ui/views/domains.py:87]: And to handle #2618 we should change the property `dkim_publickey` to return a non-splitted record.
The record then needs to be split for the zonefile.
I think I'll update your PR to reflect this.

Comment (MajliTech) [reply core/admin/mailu/ui/views/domains.py]: > And to handle #2618 we should change the property `dkim_publickey` to return a non-splitted record. The record then needs to be split for the zonefile. I think I'll update your PR to reflect this.

Go ahead, I just really wanted this to exist, as I noticed the admin panel already spit out data in this format.

**Review by nextgens — APPROVED** [review]
LGTM

**Comment by ghostwheel42** [comment]
bors try

**Review by nextgens — APPROVED** [review]

**Comment by Diman0** [comment]
bors try

**Review by Diman0 — DISMISSED** [review]
I checked the zone file using `named-checkzone` from `bind9utils`
That makes it easy to validate the zone file. 

The DKIM line from the generated zonefile has one quote too many at the end. 

Zone file created via Mailu admin:
```
example5.com. 600 IN MX 10 test.berichthier.nl.
example5.com. 600 IN TXT "v=spf1 mx a:test.berichthier.nl ~all"
dkim._domainkey.example5.com. 600 IN TXT "v=DKIM1; k=rsa; p=MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAjRIRFOP6SOT/DAhvMBiPMCmtj1iz964kUd2BiDZhtM3bPs9HSJpr559u2l3xp1igqAyKRlvHLgi60w2R5ntJBmUBe6qarG6U/dNCnT5y54XRRnL8tQ8s1FjUKRUWBnGbqTyKVEuY+alh77lvPKJGEEOpMvp/BsYVNeCjiajuYG9eAtCTv6qn6EPSy75X" "85SJADz54OqTysVeDP8WOVHc6Q+SVFo9e2YvByraiGAd2slsLX6Vo/T1AXeYkjK7T3F9vz5KuecGkRTJ18/01O6vt8DSsY5iIIu8fpKlHmgz4kgrUnmc061Ku+tmthmUo2YBaAXCrvA5PO7INohw/c1i8QIDAQAB""
_dmarc.example5.com. 600 IN TXT "v=DMARC1; p=reject; rua=mailto:admin@test.berichthier.nl; ruf=mailto:admin@test.berichthier.nl; adkim=s; aspf=s"
_imap._tcp.example5.com. 600 IN SRV 20 1 143 test.berichthier.nl.
_pop3._tcp.example5.com. 600 IN SRV 20 1 110 test.berichthier.nl.
_submission._tcp.example5.com. 600 IN SRV 20 1 587 test.berichthier.nl.
_autodiscover._tcp.example5.com. 600 IN SRV 10 1 443 test.berichthier.nl.
_submissions._tcp.example5.com. 600 IN SRV 10 1 465 test.berichthier.nl.
_imaps._tcp.example5.com. 600 IN SRV 10 1 993 test.berichthier.nl.
_pop3s._tcp.example5.com. 600 IN SRV 10 1 995 test.berichthier.nl.
autoconfig.example5.com. 600 IN CNAME test.berichthier.nl.
```

Problematic line 3:
```
dkim._domainkey.example5.com. 600 IN TXT "v=DKIM1; k=rsa; p=MIIBIjANBgkqhkiG9w0BAQEFAAOCAQ8AMIIBCgKCAQEAjRIRFOP6SOT/DAhvMBiPMCmtj1iz964kUd2BiDZhtM3bPs9HSJpr559u2l3xp1igqAyKRlvHLgi60w2R5ntJBmUBe6qarG6U/dNCnT5y54XRRnL8tQ8s1FjUKRUWBnGbqTyKVEuY+alh77lvPKJGEEOpMvp/BsYVNeCjiajuYG9eAtCTv6qn6EPSy75X" "85SJADz54OqTysVeDP8WOVHc6Q+SVFo9e2YvByraiGAd2slsLX6Vo/T1AXeYkjK7T3F9vz5KuecGkRTJ18/01O6vt8DSsY5iIIu8fpKlHmgz4kgrUnmc061Ku+tmthmUo2YBaAXCrvA5PO7INohw/c1i8QIDAQAB""
```
Note the last double quote which should not be part of the output.

File: core/admin/mailu/ui/views/domains.py [hunk 2]
```diff
@@ -70,6 +70,27 @@ def domain_details(domain_name):
...
+@ui.route('/domain/details/<domain_name>/zonefile', methods=['GET'])
+@access.domain_admin(models.Domain, 'domain_name')
+def domain_download_zonefile(domain_name):
+    domain = models.Domain.query.get(domain_name) or flask.abort(404)
+    res = [domain.dns_mx, domain.dns_spf]
+    if domain.dkim_publickey:
+        record = domain.dns_dkim.split('"', 1)[0].strip()
+        txt = f'v=DKIM1; k=rsa; p={domain.dkim_publickey}'
+        txt = ' '.join(f'"{txt[p:p+250]}"' for p in range(0, len(txt), 250))
+        res.append(f'{record} {txt}"')
```
Comment (Diman0) [inline core/admin/mailu/ui/views/domains.py:82]: I guess this is the part that must be modified. The part that introduces the unneeded extra double quote. When there is NO split, then use 
`res.append(f'{record} {txt}"')`
if there is a split, then use:
`res.append(f'{record} {txt}')`

Comment (ghostwheel42) [reply core/admin/mailu/ui/views/domains.py]: Yes and no. the quote in line 82 has to be removed for it to be correct. done.

**Comment by Diman0** [comment]
bors try

**Review by nextgens — APPROVED** [review]

**Pull request merged**